### PR TITLE
prepare support for more OSes & refactor few tests

### PR DESCRIPTION
**Pull request by mmoll**
This would bring in #184, except the FreeBSD parts in params.pp. Updated tests are still missing there, so I'd like to bring this in early to minimize diffs when other changes go in.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -4,46 +4,109 @@
...
+
+      $puppetssh_command = '/usr/bin/puppet agent --onetime --no-usecacheonfailure'
+
+      $dhcp_config = '/etc/dhcp/dhcpd.conf'
+      $dhcp_leases = '/var/lib/dhcp/dhcpd.leases'
+
+      $keyfile  = '/etc/bind/rndc.key'
+      $nsupdate = 'dnsutils'
+
+      if  ($::operatingsystem == 'Debian') and (versioncmp($::operatingsystemrelease, '8.0') >= 0) or
```
Comment (ekohl) [inline manifests/params.pp]: Alignment and the parenthesis feel weird.

Comment (mmoll) [inline manifests/params.pp]: better now? :)

File: spec/classes/foreman_proxy__plugin__openscap_spec.rb [hunk 2]
```diff
@@ -1,53 +1,57 @@
...
+      }
+    end
+
+    it 'should call the plugin' do
+      should contain_foreman_proxy__plugin('openscap')
+    end
+
+    it 'should install configuration file' do
+      should contain_foreman_proxy__settings_file('openscap')
+      content = catalogue.resource('file', '/etc/foreman-proxy/settings.d/openscap.yml').send(:parameters)[:content]
```
Comment (ekohl) [inline spec/classes/foreman_proxy__plugin__openscap_spec.rb]: While you're in here, change this to ```verify_exact_contents```?

File: spec/classes/foreman_proxy__plugin__openscap_spec.rb [hunk 3]
```diff
@@ -1,53 +1,57 @@
...
```
Comment (ekohl) [inline spec/classes/foreman_proxy__plugin__openscap_spec.rb]: Same here

File: spec/classes/foreman_proxy__plugin__openscap_spec.rb (hunk 2 above)
Comment (mmoll) [inline spec/classes/foreman_proxy__plugin__openscap_spec.rb]: sure, done!

File: manifests/plugin/chef/params.pp [hunk 4]
```diff
@@ -6,7 +6,7 @@
   $version      = undef
   $server_url   = "https://${::fqdn}"
   $client_name  = $::fqdn
-  $private_key  = '/etc/chef/client.pem'
+  $private_key  = "${foreman_proxy::etc}/chef/client.pem"
```
Comment (ekohl) [inline manifests/plugin/chef/params.pp]: This may mean that kafo needs an ```include ::foreman_proxy```.

Comment (mmoll) [inline manifests/plugin/chef/params.pp]: indeed :(

File: manifests/config.pp [hunk 5]
```diff
@@ -73,22 +73,21 @@
   if $foreman_proxy::puppetca or $foreman_proxy::puppetrun {
     if $foreman_proxy::use_sudoersd {
       if $foreman_proxy::manage_sudoersd {
-        file { '/etc/sudoers.d':
+        file { "${::foreman_proxy::sudoers}.d":
```
Comment (domcleal) [inline manifests/config.pp]: No need to quote this

File: manifests/params.pp [hunk 6]
```diff
@@ -4,46 +4,108 @@
...
+
+      $tftp_syslinux_filenames = ['/usr/share/syslinux/chain.c32',
+                                  '/usr/share/syslinux/menu.c32',
+                                  '/usr/share/syslinux/memdisk',
+                                  '/usr/share/syslinux/pxelinux.0']
+    }
+    'Debian': {
+      # if set to true, no repo will be added by this module, letting you to
+      # set it to some custom location.
+      $custom_repo         = false
```
Comment (domcleal) [inline manifests/params.pp]: Should this be outside of the conditional with the other packaging settings?

File: manifests/config.pp (hunk 5 above)
Comment (domcleal) [inline manifests/config.pp]: Oops, hadn't seen that.

**Comment by domcleal** [comment]
Thanks @mmoll, merged.

**Pull request merged**